content/docs/for-buildpack-authors/how-to/write-buildpacks/specify-env.md

```diff
@@ -3,15 +3,143 @@ title="Specify the environment"
+
+For the detect phase, the lifecycle then applies user-provided environment variables. For more information, see [clearing the buildpack environment](https://buildpacks.io/docs/for-buildpack-authors/how-to/write-buildpacks/clear-env/)
+
+For the `build` phase, the process is more complex. Before applying user-provided environment variables, the `lifecycle` applies buildpack-provided environment variables, which is anything that a previous buildpack (a buildpack that ran earlier in the `build` phase) might have configured in its `layers` directory.
+
+>Note that buildpacks cannot set environment variables for other buildpacks during the `detect` phase.
+
+#### Example
+
+Let's look at the following directory tree to see how layers created by a previous buildpack (with id `some-buildpack-id`) would affect the environment for the current buildpack.
+
+```text
+
+layers/
+└── some-buildpack-id
+    ├── some-build-layer
+    │   ├── bin
+    │   │   └── some-binary
+    │   ├── env
+    │   │   └── SOME_VAR # contents foo
+    │   └── lib
+    │       └── some-static-library
+    └── some-build-layer.toml  # has build = true in the [types] table
+
+```
+
+With this tree:
+
+* The current buildpack will see `SOME_VAR=foo` in its environment
+* The current buildpack will find `some-binary` in `PATH`
+* The current buildpack will find `some-static-library` in `LIBRARY_PATH`
+
+Thus, any `<layers>/<layer>/<env>` directory is for setting environment variables directly, and `<layers>/<layer>/<bin>`, `<layers>/<layer>/<lib>`, etc. offer a convenient way to modify `POSIX` path variables.
+
+The full list of convenience directories is summarized in the table below:
+
+| Env Variable                               | Layer Path   | Contents         | Build | Launch |
+|--------------------------------------------|--------------|------------------|-------|--------|
+| `PATH`                                     | `/bin`       | binaries         | [x]   | [x]    |
+| `LD_LIBRARY_PATH`                          | `/lib`       | shared libraries | [x]   | [x]    |
+| `LIBRARY_PATH`                             | `/lib`       | static libraries | [x]   |        |
+| `CPATH`                                    | `/include`   | header files     | [x]   |        |
+| `PKG_CONFIG_PATH`                          | `/pkgconfig` | pc files         | [x]   |        |
+
+* If `env/SOME_VAR` and `env.build/SOME_VAR` have a conflict, a given procedure applies to figure out who "wins" when applying the environment [modification rules](https://github.com/buildpacks/spec/blob/main/buildpack.md#environment-variable-modification-rules).
+* `User-defined` variables are then applied; meaning it's possible to override `buildpack-defined` variables.
+* Finally, `platform-defined` variables are applied, which eventually override any previous values.
+
+### Preparing the environment at runtime
```

Comment:
I think we can flesh out this section a little more

Reply:
could you elaborate on this? thanks

Reply:
I'm happy to land this change now, and flesh out this discussion in subsequent changes.

Reply:
Thanks @AidanDelaney

Reply:
@natalieparellano still waiting for you